src/main/java/com/suse/salt/netapi/calls/modules/State.java

```diff
@@ -18,14 +18,27 @@
     private State() { }
 
     public static LocalCall<Map<String, Object>> apply(List<String> mods) {
-        LinkedHashMap<String, Object> args = new LinkedHashMap<>();
+        return apply(mods, Optional.empty());
+    }
+
+    public static LocalCall<Map<String, Object>> apply(String... mods) {
+        return apply(Arrays.asList(mods), Optional.empty());
+    }
+
+    public static LocalCall<Map<String, Object>> apply(List<String> mods,
+            Optional<Map<String, Object>> pillar) {
+        LinkedHashMap args = new LinkedHashMap();
         args.put("mods", mods);
-        return new LocalCall<>("state.apply", Optional.empty(), Optional.of(args),
+        if (pillar.isPresent()) {
+            args.put("pillar", pillar.get());
+        }
+        return new LocalCall("state.apply", Optional.empty(), Optional.of(args),
```

Comment:
Please fix warnings throughout this code:

1. the `LinkedHashMap` should be created with type arguments: `<String, Object>`?
2. the returned `LocalCall` should be parameterized, diamond operator?